### ISSUE TITLE
Excape sequence # %% is not found if file is formatted using black

### ISSUE THREAD

**Issue by nknavasto** [sphinx-gallery/sphinx-gallery#1076]
The common python formatter [https://github.com/psf/black](url) formats the following code

```
def func():
# %%
    print("foo")
```
to 
```
def func():
    # %%
    print("foo")
```

This prevents sphinx-gallery from finding the required escape sequence in the example snippet above.
A change in the `py_source_parser.py` regular expression:
```
r'(?P<header_line>^#{20,}.*|^\# ?%%.*)\s(?P<text_content>(?:^#.*\s?)*)'
```
to
```
r'(?P<header_line>^\s*#{20,}.*|^\s*# ?%%.*)\s(?P<text_content>(?:^\s*#.*\s?)*)'
```
would fix the issue.

**Comment by larsoner** [comment]
The `# %%` end up as breaks in not just the generated HTML but also in the Jupyter notebooks. I don't think it makes sense to break in the middle of an indented section like this because it will generate invalid notebooks:
<img width="349" alt="Screenshot 2023-02-07 at 9 35 18 AM" src="https://user-images.githubusercontent.com/2365790/217274447-78ea2569-99e8-46bc-bed7-c898a312abed.png">

**Comment by nknavasto** [comment]
You are right, from a Jupyter notebook point of view the unindented comments break how the notebooks work. But looking at the issue from html, the unindented comments improve the readability considerably! Picture a use case, where you have some main function that runs multiple commands and you want to write some comments on the used commands. 
Maybe there is a way to have the best of both worlds:

1. Have nice comments that are not only python comments but are rendered as individual html blocks in between code
2. Have the compatibility to Jupyter notebooks.

This could be achieved by rendering indented `# %%` but reduce them to simple python comments when it comes to Jupyter notebook exports?

**Comment by larsoner** [comment]
To me it breaks a long standing relationship/expectation/consistency about how the code breaks work and how they relate to the notebooks, so I'm still -0.5. It's more annoying but in principle you could break your `def whatever(...)` into multiple sub-functions instead and achieve similar sectioning

**Comment by GaelVaroquaux** [comment]
> To me it breaks a long standing relationship/expectation/consistency about how the code breaks work and how they relate to the notebooks, so I'm still -0.5.

Same view.

It also breaks the ability to copy-paste from html cells.

**Comment by nknavasto** [comment]
I understand your points and I think they are very valid. In my opinion, there is still a use case for my issue, but if there is a strong coupling between the rendering of html and jupyter notebooks, then either a new kind of handling of the `# %%` needs to be looked into or this ticket can be closed. 
Thank you for your responses and feedback.
